src/pfe/file-watcher/server/src/controllers/projectsController.ts

```diff
@@ -713,18 +698,16 @@ export async function deleteProject(projectID: string): Promise<IDeleteProjectSu
             // if a delete occured for a queued build, we need to restore the rank of the rest of the builds in queue
             if (deleteQueuedBuildOccured) {
                 emitProjectRanks();
-                logger.assert((initialLength - buildQueue.length) === 1, "If project was deleted, the difference in build queue length should be exactly one");
+                logger.assert((initialLength - buildQueue.size) === 1, "If project was deleted, the difference in build queue length should be exactly one");
             }
         }, {});
 
         await lock.acquire("runningBuildsLock", async done => {
             // remove the project from in-progress queue only if the project deleted was in the in-progress build
-            runningBuilds = runningBuilds.filter((project: BuildQueueType) => {
-                if (project.operation.projectInfo.projectID === projectID) {
+            runningBuilds.forEach((buildQueueItem: BuildQueueType) => {
```

Comment:
Should we change `buildQueueItem` to `runningQueueItem` since this is looping through the running queue?

Reply:
Sure, I can rename that. I probably copied it from the `buildQueue` for each that is why it has the same name